OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -109,9 +109,14 @@ public AssetBrowserLogic(Widget widget, Action onExit, World world)
 			}
 
 			filenameInput = panel.Get<TextFieldWidget>("FILENAME_INPUT");
+			filenameInput.TakeKeyboardFocus();
 			filenameInput.OnTextEdited = () => ApplyFilter(filenameInput.Text);
-			filenameInput.OnEscKey = filenameInput.YieldKeyboardFocus;
-
+			filenameInput.OnEscKey = () =>
+			{
+				filenameInput.Text = null;
+				filenameInput.OnTextEdited();
```

Comment:
When Esc is pressed, the text field resets. So the filter has to be updated, otherwise it shows an empty text field but the list is still filtered according to the previous input.